lightning/src/routing/router.rs

```diff
@@ -1113,10 +1113,13 @@ where L::Target: Logger {
 						fees: hop.fees,
 					};
 
-					let reqd_channel_cap = if let Some (val) = final_value_msat.checked_add(match idx {
-						0 => 999,
-						_ => aggregate_next_hops_fee_msat.checked_add(999).unwrap_or(u64::max_value())
-					}) { Some( val / 1000 ) } else { break; }; // converting from msat or breaking if max ~ infinity
+					let reqd_channel_cap = if let Some (val) = final_value_msat
+						.checked_mul(ROUTE_CAPACITY_PROVISION_FACTOR).unwrap_or(u64::max_value())
```

Comment:
Does this have to all be one single statement? Can't the `match idx` be extracted into a summand variable, and perhaps the `aggregate_next_hops_fee_msat` modification, too, with some comments to accompany it?

Reply:
Cleaned it up some.